src/ProjectTemplates/Web.ProjectTemplates/content/Components-CSharp/Pages/Counter.razor

```diff
@@ -0,0 +1,27 @@
+﻿@page "/counter"
+@*#if UseServer && !UseClient -->
+@attribute [RenderModeServer]
+##endif*@
+@*#if !UseServer && UseClient -->
+@attribute [RenderModeWebassembly]
+##endif*@
+@*#if UseServer && UseClient -->
+@attribute [RenderModeAuto]
```

Comment:
What is the default for "auto" right now? i.e. how will this page render by default?

Reply:
The page first renders statically (prerenders) and emits a marker comment for both, server and webassembly. Then on the JS side of things, the JS decides whether it starts server or webassembly. That said, I am going to defer to @stevesandersonms

Reply:
Here is the spec for the `@renderMode` that @SteveSandersonMS written: https://github.com/aspnet/specs/pull/708

Reply:
> Here is the spec for the @renderMode that @SteveSandersonMS written: https://github.com/aspnet/specs/pull/708

I understand, I am putting the things that will work (as we have them right now) not the things we expect to work, especially since that depends on the compiler implementing a new directive.